Added options to cloudinary media store

stephansama opened the pull request:
<!--

Thank you for contributing to TinaCMS!

Several things must happen for your PR to be merged:

1. It must successfully build, test, and lint your branch
1. It must pass the dangerjs script
2. It must be up-to-date with master
3. It must be approved by at least 1 tinacms core member
4. It must have conventional-commits that clearly explain what has been changed


Small is beautiful! 

PRs should be small. They should be made up of many small commits, with clear 
commit messages explaining _what_ has been changed and _why_. The tests should
be short and specific, with single assertions. 

-->

Hi everyone thanks for reviewing my PR.

The purpose of the PR is to extend the cloudinary media store with a different url than the relative url currently supplied. I am currently using aws with cloudinary and tina and this is how I got it to work properly

I basically just wrapped the class in a function that accepts the options.

![Screenshot 2024-08-05 at 2 27 09 PM](https://github.com/user-attachments/assets/fcf8e6f0-5c6d-4f14-b574-d9f9eb660ed5)

Jord-Gui commented:
Hi @stephansama, could you provide some context surrounding this change (e.g. why you've made it, what issue it solves, etc.) in the description? Thanks!

stephansama commented:
> Hi @stephansama, could you provide some context surrounding this change (e.g. why you've made it/what brought about this change/what issue it solves/a link to the issue or support chat, etc.) in the description? Thanks!

:facepalm: sorry about that i have updated it thank you for commenting!

kldavis4 commented:
@stephansama there seems to be a lot more than just the requested functionality

In order to make it easier to review, please limit your changes to only those necessary

stephansama commented:
> @stephansama there seems to be a lot more than just the requested functionality
> 
> In order to make it easier to review, please limit your changes to only those necessary



> @stephansama there seems to be a lot more than just the requested functionality
> 
> In order to make it easier to review, please limit your changes to only those necessary

Hey Kelly thanks for reviewing my PR. However, I'm not sure I understand what you mean?

My changes are as minimal as possible and only modify the adapter for the use case i outlined.

I wrapped the existing class in a factory function that accepts the new options I added.
and I added a file with the type definitions for the options.
I didn't change anything else about the in place implementation.

kldavis4 commented:
> > @stephansama there seems to be a lot more than just the requested functionality
> > In order to make it easier to review, please limit your changes to only those necessary
> 
> Hey @kldavis4 thanks for commenting on my PR. However, I'm not sure I understand what you mean?
> 
> My changes are as minimal as possible and only modify the adapter for the use case i outlined.
> 
> I wrapped the existing class in a factory function that accepts the new options I added. I made one function that was private [`buildQuery`] public because i was receiving errors creating a class with a private function. and I added a file with the type definitions for the options. I didn't change anything else about the in place implementation.

I think the whitespace changes are causing a bigger change than described above

kldavis4 commented on packages/next-tinacms-cloudinary/src/cloudinary-media-store.ts:
```diff
@@ -1,3 +1,4 @@
+import type { TinaCloudCloudinaryOptions } from './options'
```
define and export this type here instead of `options.ts`

kldavis4 commented on packages/next-tinacms-cloudinary/src/cloudinary-media-store.ts:
```diff
@@ -6,105 +7,104 @@ import type {
-    return fetch(input, init)
-  }
-  accept = DEFAULT_MEDIA_UPLOAD_TYPES
+export const createCloudinaryMediaStore = (
```
I don't think we need the factory method here

kldavis4 commented on packages/next-tinacms-cloudinary/src/cloudinary-media-store.ts:
```diff
@@ -6,105 +7,104 @@ import type {
-        },
-        src: fileRes.secure_url,
-      }
+        const res = await this.fetchFunction(options.baseURL, {
```
whitespace

kldavis4 commented on packages/next-tinacms-cloudinary/src/cloudinary-media-store.ts:
```diff
@@ -6,105 +7,104 @@ import type {
-      const error = interpretErrorMessage(e)
-      throw error
+    async delete(media: Media) {
+      await this.fetchFunction(options.baseURL + encodeURIComponent(media.id), {
```
keep the template literal

kldavis4 commented on packages/next-tinacms-cloudinary/src/cloudinary-media-store.ts:
```diff
@@ -6,105 +7,104 @@ import type {
-      nextOffset: offset,
+    async list(options: MediaListOptions): Promise<MediaList> {
+      const query = this.buildQuery(options)
+      const response = await this.fetchFunction(options.baseURL + query)
```
whitespace

kldavis4 commented on packages/next-tinacms-cloudinary/src/cloudinary-media-store.ts:
```diff
@@ -6,105 +7,104 @@ import type {
-      .filter((key) => options[key] !== '' && options[key] !== undefined)
-      .map((key) => `${key}=${options[key]}`)
-      .join('&')
+    buildQuery(options: MediaListOptions) {
```
whitespace

kldavis4 commented on packages/next-tinacms-cloudinary/src/options.ts:
```diff
@@ -0,0 +1 @@
+export type TinaCloudCloudinaryOptions = { baseURL: string }
```
`baseURL` -> `baseUrl`

```suggestion
export type TinaCloudCloudinaryOptions = { baseUrl: string }
```

kldavis4 commented on packages/next-tinacms-cloudinary/src/cloudinary-tina-cloud-media-store.ts:
```diff
@@ -1,27 +1,29 @@
-        const query = `${url.includes('?') ? '&' : '?'}clientID=${
-          client.clientId
-        }`
+export const createTinaCloudinaryMediaStore = (
```
```suggestion
export const createTinaCloudCloudinaryMediaStore = (
```

kldavis4 commented:
I created a new PR based on your changes (and my comments) here: https://github.com/tinacms/tinacms/pull/4704 and published a tag version https://www.npmjs.com/package/next-tinacms-cloudinary/v/0.0.0-20240807010824 so that you can test the changes. Please give it a try and let me know whether it solves your problem.

stephansama commented:
> I created a new PR based on your changes (and my comments) here: #4704 and published a tag version https://www.npmjs.com/package/next-tinacms-cloudinary/v/0.0.0-20240807010824 so that you can test the changes. Please give it a try and let me know whether it solves your problem.

I just installed the package that u published.
Your implementation is working well and maintains better backwards compatibility than mine. I appreciate you checking it out and adding your comments. I will remember these lessons for the future ❤️💛💚

kldavis4 commented on packages/next-tinacms-cloudinary/src/cloudinary-media-store.ts:
```diff
@@ -66,16 +72,13 @@ export class CloudinaryMediaStore implements MediaStore {
-        method: 'DELETE',
-      }
-    )
+    await this.fetchFunction(this.baseUrl + encodeURIComponent(media.id), {
```
```suggestion
    await this.fetchFunction(`${this.baseUrl}/${encodeURIComponent(media.id)}`, {
```

kldavis4 reviewed (dismissed):
Looks good - thanks for the contribution

stephansama commented:
> Looks good - thanks for the contribution

thanks for the advice from discord. looking over my pr and giving me the opportunity to contribute i really appreciate it ❤️💛💚

stephansama commented:
oops looks like i might need to updated it again

kldavis4 commented on packages/next-tinacms-cloudinary/src/cloudinary-media-store.ts:
```diff
@@ -66,16 +72,13 @@ export class CloudinaryMediaStore implements MediaStore {
   async list(options: MediaListOptions): Promise<MediaList> {
     const query = this.buildQuery(options)
-    const response = await this.fetchFunction('/api/cloudinary/media' + query)
+    const response = await this.fetchFunction(options.baseURL + query)
```
```suggestion
    const response = await this.fetchFunction(options.baseUrl + query)
```

kldavis4 commented on packages/next-tinacms-cloudinary/src/cloudinary-media-store.ts:
```diff
@@ -66,16 +72,13 @@ export class CloudinaryMediaStore implements MediaStore {
   async list(options: MediaListOptions): Promise<MediaList> {
     const query = this.buildQuery(options)
-    const response = await this.fetchFunction('/api/cloudinary/media' + query)
+    const response = await this.fetchFunction(options.baseUrl + query)
```
```suggestion
    const response = await this.fetchFunction(this.baseUrl + query)
```
